pkg/debug/diff.go

```diff
@@ -0,0 +1,106 @@
+package debug
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sort"
+	"strings"
+
+	"github.com/google/go-github/v31/github"
+	"github.com/kylelemons/godebug/diff"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// GenerateDiff attempts to pull a dependency list from a previous job (job, jobID) and generate a diff against a provided string
+func GenerateDiff(jobName, jobID, phase, dependencies string) (string, error) {
+	resp, err := http.Get(fmt.Sprintf("https://storage.googleapis.com/origin-ci-test/logs/%s/%s/artifacts/%s/dependencies.txt", jobName, jobID, phase))
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	return diff.Diff(string(body), dependencies), nil
+}
+
+// GenerateDependencies creates a list of images and the MCC hash
+func GenerateDependencies(kube kubernetes.Interface) (dependencies string, err error) {
+	pods, err := kube.CoreV1().Pods("").List(metav1.ListOptions{})
+	data := []string{}
+	if err != nil {
+		return "", err
+	}
+	images, err := GetImageList(pods)
+	if err != nil {
+		return "", err
+	}
+	hash, err := GetCurrentMCCHash()
+	if err != nil {
+		return "", err
+	}
+
+	dependencies = "MCC: " + hash + "\n-----\n"
+
+	for k, v := range images {
+		data = append(data, fmt.Sprintf("%-80s: %s", v, k))
+	}
+
+	sort.Strings(data)
+	dependencies += strings.Join(data, "\n")
+
+	return dependencies, nil
```

Comment:
I feel like we may need to sort this list somehow -- if it's already sorted that's fine, but just so we know we're comparing apples to apples.

Reply:
`sort.Strings(data)` is already sorting it. 

Here's an example of the output:

```
➜  osde2e git:(SDCICD-273) cat /tmp/osde2e939856810/install/dependencies.txt
MCC: 4f6fca8a086050201d6ff4160de9cdd4fb2ccb26
-----
Unknown/certified-operators                                                     : quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:c2d121db068583a6dc84336c3d9753172c830bad1498d064fd4a7abc7726bc00
Unknown/dns                                                                     : quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:6d6df0556a250b7e7b58b37666baf75b688080ddfac5c8622eae3c7d586c6dd2
Unknown/dns-node-resolver                                                       : quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:c23dd75611fa69829a5e27281efca2a82ba1b9a9ce8a77d19264febf31df19ea
Unknown/registry                                                                : quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:3614c1770847d9ed32ebe3f33a9a05b1749261daf191e64700011824418eeb23
Unknown/router                                                                  : quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:a4263ebc8f03c9ee629a23892afcdc450328e05c3b42717dc1c533a9c317a8fc
Unknown/tuned                                                                   : quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e0505e2fab966054f3d3bd567a62f5e07c7f6ab984e963cc5a490e4713971d52
authentication-operator/operator                                                : quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:d3369e7d4a78592c65e926c75a8c9e2030eb0ab0c0fae3ccbf973359eaa60c82
catalog-operator/catalog-operator                                               : quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:cb3b16723b71f6157060205f59fb5601f5d45a6da6acbda31ce8fef7a9c5f2a8
cloud-credential-operator/manager                                               : quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e81a86dff64a474f4d93436537103bb55370aafee394d171505faee3a2f80a95
cluster-image-registry-operator/cluster-image-registry-operator                 : quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:8d77e61765295b942a300f337c1c7310d09bc8b4baa67513bf827dfd2e0a5bd2
cluster-node-tuning-operator/cluster-node-tuning-operator                       : quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:2006a5ef6cab527d7867551d412066f12253f7cd40ba2834157bd8fadeaf68af
cluster-samples-operator/cluster-samples-operator                               : quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:2e6e72f1b09120c5421d34a0380e39a6472e5cc9c4bcaeaa32e1e0973fce1d2d
cluster-storage-operator/cluster-storage-operator                               : quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:14aa9f25ac3dd0bc09fac65f503ba60317525819a9850dd66cd6c4f1ea324c5b
cluster-version-operator/cluster-version-operator                               : quay.io/openshift-release-dev/ocp-release@sha256:edb4364367cff4f751ffdc032bc830a469548f998127b523047a8dd518c472cd
console-operator/console-operator                                               : quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:f5f6c0e477ccb77900f1670d56ab4bf2778bde8ed2b82d33127cab65100251f3
console/console                                                                 : quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:0f6c50788cce8b3130d9693c6941968a5cef24df6b20cb83b17c05076f1242df
console/download-server                                                         : quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:4c2139ab831a11d6c706977176fe25d5a24903d4a0ae90735d3df9849a24c88d
dns-operator/dns-operator                                                       : quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:9e40b664842371de015a57169e3e20dda757524766c02ba4d790467137cd2b79
```

Reply:
I just glazed over strings.Sort... whoops.